src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -13,24 +13,72 @@
 
     private NetworkTableEntry yaw = table.getEntry("/tx");
     private NetworkTableEntry pitch = table.getEntry("/ty");
-    private NetworkTableEntry lightMode = table.getEntry("/ledMode");
+    private NetworkTableEntry ledMode = table.getEntry("/ledMode");
     private NetworkTableEntry validTarget = table.getEntry("/tv");
     private NetworkTableEntry skew = table.getEntry("/ts");
     private NetworkTableEntry cameraMode = table.getEntry("/camMode");
     private NetworkTableEntry pipeline = table.getEntry("/pipeline");
     private NetworkTableEntry targetState = table.getEntry("/target_state");
     private NetworkTableEntry poseData = table.getEntry("/camtran");
 
-    public static class Constants {
+    public static enum LEDMode {
         // LED MODES /limelight/ledMode
-        public static final int LED_MODE_FROM_PIPELINE = 0;
-        public static final int LED_MODE_FORCE_OFF = 1;
-        public static final int LED_MODE_FORCE_BLINK = 2;
-        public static final int LED_MODE_FORCE_ON = 3;
+        FROM_PIPELINE(0),
+        FORCE_OFF(1),
+        FORCE_BLINK(2),
+        FORCE_ON(3);
 
-        // pipeline numbers
-        public static final int PIPELINE_TELEOP = 0;
-        public static final int PIPELINE_GET_POS = 1;
+        private int ledMode;
+
+        private LEDMode(int ledMode) {
+            this.ledMode = ledMode;
+        }
+
+        public int intValue() {
+            return ledMode;
+        }
+        
+        public static LEDMode valueOf(int ledMode) {
+            switch (ledMode) {
+                case 0:
+                    return FROM_PIPELINE;
+                case 1:
+                    return FORCE_OFF;
+                case 2:
+                    return FORCE_BLINK;
+                case 3:
+                    return FORCE_ON;
+                default:
+                    throw new IllegalArgumentException("Unknown LED mode.");
+            }
+        }
+    }
+
+    public static enum Pipeline {
+        // pipeline numbers /limelight/pipeline
+        TELEOP(0),
+        GET_POS(1);
+
+        private int pipeline;
+
+        private Pipeline(int pipeline) {
+            this.pipeline = pipeline;
+        }
+
+        public int intValue() {
+            return pipeline;
+        }
+
+        public static Pipeline valueOf(int pipeline) {
+            switch (pipeline) {
+                case 0:
+                    return TELEOP;
+                case 1:
+                    return GET_POS;
+                default:
+                    throw new IllegalArgumentException("Unknown pipeline number.");
```

Comment:
Now that I look at it, it's possible that the limelight may set its network tables value `limelight/pipeline` to something other than 0 or 1 and this method would cause an unexpected `IllegalArgumentException`. This would be from us creating a pipeline without doing it in the code, which is definitely possible. To prevent a crash from this is might be good to make a `UNKNOWN` enum constant for the `Pipeline` enum. What do you folks think?

Reply:
Could you use `None`?

Reply:
I think `None` conveys the idea that a pipeline hasn't been set, when that isn't really the case